microbench/bfs/src/sssp.cpp

```diff
@@ -12,9 +12,7 @@ void bfs::updateData(std::uint64_t val, pando::GlobalRef<std::uint64_t> ref) {
     if (val >= temp) {
       break;
     }
-  } while (!pando::atomicCompareExchange(&ref, pando::GlobalPtr<std::uint64_t>(&temp),
-                                         pando::GlobalPtr<std::uint64_t>(&val),
-                                         std::memory_order_relaxed, std::memory_order_relaxed));
+  } while (!pando::atomicCompareExchange(&ref, temp, val));
```

Comment:
please explain why this change is correct?

Reply:
would prefer if this change was separate PR, also why is memory_order_relaxed wrong?

Reply:
There is zero reason I can understand that this could work.

Reply:
They are basically the exact same code. Just like Max said in Discord, usual atomic operations should only contain one pointer or address and the rest operand should be scalar numbers. There were already multiple versions of atomicCompareExchange, one takes GlobalPtr as operands and the other takes scalar as operands. I figured out there is no reason for me to intentionally transfom from scalar to GlobalPtr in this case here.

Reply:
There is nothing wrong with memory_ordered_relaxed. It's just that the scalar operand version does no take memory order as argument. The default order is __ATOMIC_SEQ_CST (total ordering) by the function.